backend/service_accounts/authentication.py

```
@@ -0,0 +1,58 @@
+            plaintext = header[1].decode('utf-8')
+        except UnicodeDecodeError:
+            return None
+        if not plaintext.startswith('sa_'):
+            return None
+
+        try:
+            token = ServiceAccountToken.objects.select_related(
+                'service_account'
+            ).get(digest=ServiceAccountToken.hash_token(plaintext))
+        except ServiceAccountToken.DoesNotExist:
+            raise AuthenticationFailed('Invalid service account token.')
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)